ui/src/app/lib/atlasmap-data-mapper/components/line-machine.component.ts

```diff
@@ -50,13 +50,21 @@ export class LineMachineComponent implements OnInit {
 
   @ViewChild('lineMachineElement') lineMachineElement: ElementRef;
 
-  private yOffset = 3;
+  redrawMappingLinesEventListener = document.addEventListener('redrawMappingLines', function(e: CustomEvent) {
+      const lmcInstance: LineMachineComponent = e.detail;
+      setTimeout(() => {
+          lmcInstance.redrawLinesForMappings();
+          }, 10);
+    }, false);
```

Comment:
We’re mixing native DOM API methods and globals (document, for instance), with callbacks as functions, assignment expressions with typing, parameters and constant names not matching its type, 4 spaces indenting, and defer execution using an arbitrary value (10).

This block might require some love and leverage Angular’s built in API for emitting events instead